app/helpers/form_helper.py

```diff
@@ -0,0 +1,109 @@
+from app.helpers.schema_helper import SchemaHelper
+from app.forms.household_composition_form import generate_household_composition_form, deserialise_composition_answers
+from app.forms.household_relationship_form import build_relationship_choices, generate_relationship_form
+from app.forms.questionnaire_form import generate_form
+
+
+def get_form_for_location(block_json, location, answer_store, error_messages, disable_mandatory=False):
```

Comment:
This feels like multiple functions in one ; might be worth splitting each of the if clauses into a separate function since each block of code behaves so differently; the flag for disable mandatory also looks like another toggle to the function which could be handled by the caller to reduce the complexity/variants through this code.